### PR TITLE
[office-js, office-js-preview] Removing HTML tables from requirement set and host descriptions

### PR DESCRIPTION
**Pull request by AlexJerabek**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.microsoft.com/en-us/office/dev/add-ins/reference/requirement-sets/office-add-in-requirement-sets
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by AlexJerabek** [comment]
Please do not merge without another maintainer's approval.

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -803,10 +552,9 @@ declare namespace Office {
          * [Api set: Mailbox 1.0]
          *
          * @remarks
-         * <table>
-         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}</td><td>Restricted</td></tr>
-         *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td><td>Compose or Read</td></tr>
-         * </table>
+         * 
+         * **{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}:** Restricted
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]:
```suggestion
         * **{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}**: Restricted
```

Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: I would prefer all of these to exclude the colon from being bolded

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -953,27 +653,12 @@ declare namespace Office {
              *
              * @remarks
              * 
-             * <table>
-             *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}</td><td>Restricted</td></tr>
-             *   <tr><td>{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}</td><td>Compose or Read</td></tr>
-             * </table>
-             * 
-             * **Support details**
+             * **{@link https://docs.microsoft.com/outlook/add-ins/understanding-outlook-add-in-permissions | Minimum permission level}:** Restricted
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]: Hmm, this prop is weird. By keeping just the Outlook info, it seems like it's only supported by Outlook and not the other hosts.

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -953,27 +653,12 @@ declare namespace Office {
...
+             * **{@link https://docs.microsoft.com/outlook/add-ins/#extension-points | Applicable Outlook mode}:** Compose or Read
              * 
-             * A capital Y in the following matrix indicates that this method is supported in the corresponding Office host application. 
-             * An empty cell indicates that the Office host application doesn't support this method.
+             * This property is only supported in Outlook in [requirement set](/office/dev/add-ins/reference/requirement-sets/outlook-api-requirement-sets) Mailbox 1.3 and later.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]:
```suggestion
             * This property is introduced in Outlook in [requirement set](/office/dev/add-ins/reference/requirement-sets/outlook-api-requirement-sets) Mailbox 1.3 but without the `options` parameter. Support for `options` is added in Mailbox Preview.
```

File: types/office-js-preview/index.d.ts [hunk 4]
```diff
@@ -953,27 +653,12 @@ declare namespace Office {
...
-             *   <tr><td><strong> Excel      </strong></td><td> Y                                                                                                           </td><td> Y                          </td><td> Y               </td></tr>
-             *   <tr><td><strong> Outlook    </strong></td><td> Y (Since Mailbox 1.3: without `options` parameter;<br>Mailbox Preview: adds support for `options` parameter)</td><td>                            </td><td>                 </td></tr>
-             *   <tr><td><strong> PowerPoint </strong></td><td> Y                                                                                                           </td><td> Y                          </td><td> Y               </td></tr>
-             *   <tr><td><strong> Word       </strong></td><td> Y                                                                                                           </td><td> Y                          </td><td> Y               </td></tr>
-             *  </table>
+             * Mailbox 1.3 does not have the `options` parameter while Mailbox Preview does have support for `options` parameter)
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]: If you accept the suggestion for the previous comment, then remove this sentence.

File: types/office-js-preview/index.d.ts [hunk 5]
```diff
@@ -1107,10 +792,10 @@ declare namespace Office {
         * Displays a dialog to show or collect information from the user or to facilitate Web navigation.
         *
         * @remarks
-        * <table>
-        *   <tr><td>Hosts</td><td>Word, Excel, Outlook, PowerPoint</td></tr>
-        *   <tr><td>Requirement sets</td><td>DialogApi, Mailbox 1.4</td></tr>
-        * </table>
+        * **Hosts:** >Word, Excel, Outlook, PowerPoint
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]:
```suggestion
        * **Hosts**: Word, Excel, Outlook, PowerPoint
```

File: types/office-js-preview/index.d.ts [hunk 6]
```diff
@@ -1946,23 +1535,6 @@ declare namespace Office {
     }
     /**
      * Specifies the type of the XML node.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]: Shouldn't this one include pointing folks to the section on the CustomXmlParts reqt. set?

File: types/office-js-preview/index.d.ts [hunk 7]
```diff
@@ -2429,36 +1801,15 @@ declare namespace Office {
     /**
     * Represents a binding to a section of the document.
     *
-    * @remarks
-    * <table><tr><td>Requirement Sets</td><td>MatrixBindings, TableBindings, TextBindings</td></tr></table>
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]: Keep with links to appropriate sections on SOT?

File: types/office-js-preview/index.d.ts [hunk 8]
```diff
@@ -2500,7 +1857,15 @@ declare namespace Office {
          * Returns the data contained within the binding.
          *
          * @remarks
-         * <table><tr><td>Requirement Sets</td><td>MatrixBindings, TableBindings, TextBindings</td></tr></table>
+         * **Requirement Sets:**
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]: Does this need all the coercion items?

File: types/office-js-preview/index.d.ts [hunk 9]
```diff
@@ -2959,7 +2317,9 @@ declare namespace Office {
          * Creates a binding against a named object in the document.
          *
          * @remarks
-         * <table><tr><td>Requirement Sets</td><td>MatrixBindings, TableBindings, TextBindings</td></tr></table>
+         * {@link https://docs.microsoft.com/office/dev/add-ins/reference/requirement-sets/office-add-in-requirement-sets#matrixbindings | MatrixBindings},
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]: Should include Requirement sets "header"

File: types/office-js-preview/index.d.ts [hunk 10]
```diff
@@ -3136,23 +2425,10 @@ declare namespace Office {
...
-         * {@link https://docs.microsoft.com/office/dev/add-ins/concepts/requirements-for-running-office-add-ins | Requirements for running Office Add-ins}.
-         * 
-         * *Supported hosts, by platform*
-         *  <table>
-         *   <tr><th>                         </th><th> Office for Windows desktop </th><th> Office Online (in browser) </th><th> Office for iPad </th><th> Office for Mac </th></tr>
-         *   <tr><td><strong> Access </strong></td><td>                            </td><td> Y                          </td><td>                 </td><td>                </td></tr>
-         *   <tr><td><strong> Excel  </strong></td><td> Y                          </td><td> Y                          </td><td> Y               </td><td> Y              </td></tr>
-         *   <tr><td><strong> Word   </strong></td><td> Y                          </td><td> Y                          </td><td> Y               </td><td> Y              </td></tr>
-         *  </table>
+         * **Requirement Sets:**
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]:
```suggestion
         * **Requirement sets** 
```

File: types/office-js-preview/index.d.ts [hunk 11]
```diff
@@ -3803,6 +2942,9 @@ declare namespace Office {
         getByNamespaceAsync(ns: string, options?: Office.AsyncContextOptions, callback?: (result: AsyncResult<CustomXmlPart[]>) => void): void;
         /**
          * Asynchronously gets the specified custom XML part(s) by its namespace.
+         * 
+         * @remarks
+         * **Requirement Sets:** [CustomXmlParts](/office/dev/add-ins/reference/requirement-sets/office-add-in-requirement-sets#customxmlparts)
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]:
```suggestion
         * **Requirement sets**: [CustomXmlParts](/office/dev/add-ins/reference/requirement-sets/office-add-in-requirement-sets#customxmlparts)
```

File: types/office-js-preview/index.d.ts [hunk 12]
```diff
@@ -4202,8 +3173,10 @@ declare namespace Office {
          * Note that specifying file slice size of above permitted limit will result in an "Internal Error" failure.
          *
          * @remarks
-         * <table><tr><td>Requirement Sets</td><td>File</td></tr></table>
-         *
+         * {@link https://docs.microsoft.com/office/dev/add-ins/reference/requirement-sets/office-add-in-requirement-sets#compressedfile | CompressedFile} (when using `Office.FileType.Compressed`),
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]: missing Requirement sets "header"

**Review by ElizabethSamuel-MSFT — CHANGES_REQUESTED** [review]
These are my comments so far.

File: types/office-js-preview/index.d.ts (hunk 2 above)
Comment (AlexJerabek) [reply types/office-js-preview/index.d.ts]: Follow up: is the Source property only an Outlook thing? Or was that table inaccurate?

File: types/office-js-preview/index.d.ts (hunk 8 above)
Comment (AlexJerabek) [reply types/office-js-preview/index.d.ts]: They are called out in the SoT conceptual page for this method.  Again, this is just my logic. I'm open to a better system.

File: types/office-js-preview/index.d.ts (hunk 2 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: I can't say if source is Outlook only. Looks that way.

File: types/office-js-preview/index.d.ts [hunk 13]
```diff
@@ -670,24 +490,10 @@ declare namespace Office {
...
-        * **Support details**
-        * 
-        * A capital Y in the following matrix indicates that this property is supported in the corresponding Office host application. 
-        * An empty cell indicates that the Office host application doesn't support this enumeration.
-        * 
-        * For more information about Office host application and server requirements, see 
-        * {@link https://docs.microsoft.com/office/dev/add-ins/concepts/requirements-for-running-office-add-ins | Requirements for running Office Add-ins}.
-        * 
-        * *Supported hosts, by platform*
-        *  <table>
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts]: Where else can a user get this info?

File: types/office-js-preview/index.d.ts (hunk 8 above)
Comment (ElizabethSamuel-MSFT) [reply types/office-js-preview/index.d.ts]: ok

**Comment by AlexJerabek** [comment]
Thanks for the review @ElizabethSamuel-MSFT. A new version has been committed.

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
@AlexJerabek ready to merge?

**Comment by AlexJerabek** [comment]
@uniqueiniquity Go to go. Thanks!

**Pull request merged**